mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/SocialScreen.kt

```diff
@@ -28,97 +33,135 @@ import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Searching
  * @param P the type of the [Person].
  * @param state the [SocialScreenState], manage the composable contents.
  * @param modifier the [Modifier] for the composable.
+ * @param key a function which uniquely identifies the list items.
+ * @param contentPadding the [PaddingValues] for this screen.
  */
 @OptIn(ExperimentalAnimationApi::class)
 @Composable
-fun <P : Person> SocialScreen(state: SocialScreenState<P>, modifier: Modifier = Modifier) {
-
-  val strings = LocalLocalizedStrings.current
-
+fun <P : Person> SocialScreen(
+    state: SocialScreenState<P>,
+    modifier: Modifier = Modifier,
+    key: ((P) -> Any)? = null,
```

Comment:
[Question] Why use a function to uniquely identify the list items?

Reply:
This is required to support animated items. Keys help Compose associate the state of each item with its associated composable : https://developer.android.com/jetpack/compose/lists#item-keys.